bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java

```diff
@@ -115,6 +171,11 @@ public void added(ItemChannelLink itemChannelLink) {
 
         @Override
         public void removed(ItemChannelLink itemChannelLink) {
+            if (itemRegistry.get(itemChannelLink.getItemName()) == null) {
+                // Don't inform about the unlink if the item does not exist.
+                // The handler is informed if the item has been removed.
+                return;
```

Comment:
> I would be fine to actually only call channelUnlinked() when a link is removed and don't care about items.

Okay

> The only thing that could happen is that an item disappears unnoticed and that the handler tries to send a command/update nonetheless. But as this can anyhow happen anytime (since hardly any handler first checks via "isLinked()", I don't think that there is any issue with that.

The test that recently fails needs at least one call to the unlink method and does not succeed if there is no such call.
So, I will ignore the existence of the item if a link is removed.